dali/operators/geometry/mt_transform_attr.cc

```diff
@@ -0,0 +1,183 @@
+different number of components.
+
+If a scalar value is provided, ``M`` is assumed to be a square matrix with that value on the
+diagonal. The size of the matrix is then assumed to match the number of components in the
+input vectors.)",
+    nullptr,  // no default value
+    true)
+  .AddOptionalArg<vector<float>>("T", R"(The translation vector.
+
+If left unspecified, no translation is applied.
+
+The number of components of this vector must match the number of rows in matrix ``M``.
+If a scalar value is provided, that value is broadcast to all components of ``T`` and the number
+of components is chosen to match the number of rows in ``M``.)", nullptr, true)
+  .AddOptionalArg<vector<float>>("MT", R"(A block matrix [M T] which combines the arguments
+``M`` and ``T``.
+
+Providing a scalar value for this argument is equivalent to providing the same scalar for
+M and leaving T unspecified.
+
+The number of columns must be one more than the number of components in the input.
+This argument is mutually exclusive with ``M`` and ``T``.)",
+    nullptr,
+    true);
+
+void MTTransformAttr::ProcessMatrixArg(const OpSpec &spec, const ArgumentWorkspace &ws, int N) {
+  bool is_fused = HasFusedMT();
+  const char *name = is_fused ? "MT" : "M";
+
+  int cols = input_pt_dim_ + (is_fused ? 1 : 0);
+  if (is_fused)
+    translation_.clear();
+
+  if (HasMatrixRegularArg()) {
+    mtx_ = spec.template GetRepeatedArgument<float>(name);
+    if (mtx_.size() == 1) {
+      SetOutputPtDim(input_pt_dim_);
+      mtx_.resize(output_pt_dim_ * input_pt_dim_);
+      FillDiag(mtx_, mtx_[0]);
+      if (is_fused) {
+        translation_.resize(output_pt_dim_, 0);
+      }
+    } else {
+      DALI_ENFORCE(mtx_.size() % cols == 0, make_string("Cannot form a matrix ",
+        mtx_.size(), " elements and ", cols, "columns"));
+      SetOutputPtDim(mtx_.size() / cols);
+      if (is_fused)
+        SplitFusedMT();
+    }
+    Repeat(per_sample_mtx_, mtx_, N);
+    if (is_fused)
+      Repeat(per_sample_translation_, translation_, N);
+  } else if (HasMatrixInputArg()) {
+    const auto &M_inp = ws.ArgumentInput(name);
+    auto M = view<const float>(M_inp);
+    DALI_ENFORCE(is_uniform(M.shape), "Matrices for all samples int the batch must have "
+      "the same shape.");
+    DALI_ENFORCE(M.shape.sample_dim() == 0 || M.shape.sample_dim() == 2, make_string(
+      "The parameter M must be a list of 2D matrices of same size or a list of scalars. Got: ",
+      M.shape));
+    if (M.shape.sample_dim() == 0) {  // we have a list of scalars - put them on diagonals
+      SetOutputPtDim(cols);
+      int mat_size = output_pt_dim_ * cols;
+      per_sample_mtx_.resize(mat_size * N);
+      for (int i = 0; i < N; i++) {
+        FillDiag(make_span(&per_sample_mtx_[i * mat_size], mat_size), M.data[i][0]);
+      }
+      translation_.resize(output_pt_dim_, 0);
+      Repeat(per_sample_translation_, translation_, N);
```

Comment:
Can add a comment why the `if (is_fused)` is not relevant here?

Reply:
It's doesn't change anything, but it should be here to avoid a redundant operation.

Reply:
I know, but for consistency I would leave at least a comment that it is skipped on purpose here.

Reply:
What I meant is: the `if` should be here. I added it.